Source/Core/Core/IOS/ES/TitleManagement.cpp

```diff
@@ -350,7 +350,25 @@ IPCCommandResult ES::ImportContentEnd(Context& context, const IOCtlVRequest& req
 
 ReturnCode ES::ImportTitleDone(Context& context)
 {
-  if (!context.title_import.tmd.IsValid())
+  if (!context.title_import.tmd.IsValid() || context.title_import.content_id != 0xFFFFFFFF)
+    return ES_EINVAL;
+
+  // Make sure all listed, non-optional contents have been imported.
+  const u64 title_id = context.title_import.tmd.GetTitleId();
+  const std::vector<IOS::ES::Content> contents = context.title_import.tmd.GetContents();
+  const IOS::ES::SharedContentMap shared_content_map{Common::FROM_SESSION_ROOT};
+  const bool has_all_required_contents =
+      std::all_of(contents.begin(), contents.end(), [&](const IOS::ES::Content& content) {
```

Comment:
Hmm, I'm not sure when cbegin/cend should be used (vs begin/end). Is it like `const` where you should use it everywhere you can (without making the code too hard to read)? Or only when you want to be sure to get a const_iterator instead of an iterator?